packages/kit/test/apps/basics/src/routes/routing/_tests.js

```diff
@@ -188,15 +188,64 @@ export default function (test, is_dev) {
 	test(
 		'back button returns to previous route when previous route has been navigated to via hash anchor',
 		'/routing/hashes/a',
-		async ({ page, clicknav, back }) => {
-			await clicknav('[href="#hash-target"]');
+		async ({ page, clicknav }) => {
+			await page.click('[href="#hash-target"]');
 			await clicknav('[href="/routing/hashes/b"]');
 
-			await back();
+			await page.goBack();
 			assert.equal(await page.textContent('h1'), 'a');
 		}
 	);
 
+	test(
+		'focus works if page load has hash',
+		'/routing/hashes/target#p2',
+		async ({ page }) => {
+			await page.keyboard.press('Tab');
+			assert.equal(
+				await page.evaluate(() => (document.activeElement || {}).textContent),
+				'next focus element'
+			);
+		},
+		// Does not work in nojs
+		{ nojs: false }
```

Comment:
In no-js, the default page focus after `Tab` is the document body. I'm not entirely clear why it works in js, but I'm skipping this for now.